src/codegen/ast.lisp

```diff
@@ -102,9 +105,15 @@
 (defstruct (node-abstraction
             (:include node)
             (:constructor node-abstraction (type vars subexpr)))
-  (vars    (required 'vars)    :type symbol-list :read-only t)
+  (vars    (required 'vars)    :type symbol-list        :read-only t)
   (subexpr (required 'subexpr) :type node               :read-only t))
 
+(defstruct (node-bare-abstraction
+            (:include node)
+            (:constructor node-bare-abstraction (type vars subexpr)))
```

Comment:
can you add a doc string to abstraction and bare-abstraction (and maybe others too)?